Integer overflow in `RobustListHead::futex_addr()`

### Describe the bug
There is a integer overflow in `RobustListHead::futex_addr()` at kernel/src/process/posix_thread/robust_list.rs:53.

https://github.com/asterinas/asterinas/blob/f01772ca853e76d3076a561da3281034e3a46196/kernel/src/process/posix_thread/robust_list.rs#L53

### To Reproduce
Compile and run the PoC program:
```C
#define _GNU_SOURCE
#include <linux/futex.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/syscall.h>
#include <unistd.h>

int main() {
  struct robust_list_head head;
  head.list.next = (struct robust_list *)(uintptr_t)(0x7fffffffffffffff);
  head.futex_offset = 16;
  head.list_op_pending = NULL;

  syscall(SYS_set_robust_list, &head, sizeof(head));
  perror("SYS_set_robust_list");

  syscall(SYS_exit_group, 0);
  perror("SYS_exit_group");
}
```

### Expected behavior

Asterinas reports add with overflow and is terminated.

### Environment
- Official docker asterinas/asterinas:0.8.1
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 8bfbdf6

### Logs
```
panicked at /root/asterinas/kernel/src/process/posix_thread/robust_list.rs:53:9:
attempt to add with overflow
Printing stack trace:
   1: fn 0xffffffff887e5be0 - pc 0xffffffff887e5c4a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c829110;

   2: fn 0xffffffff887e59f0 - pc 0xffffffff887e5b40 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c829170;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c8292c0;

   4: fn 0xffffffff889c3240 - pc 0xffffffff889c32c2 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c8292d0;

   5: fn 0xffffffff889a0820 - pc 0xffffffff889a084e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c829360;

   6: fn 0xffffffff884926a0 - pc 0xffffffff884926d0 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c8293b0;

   7: fn 0xffffffff88492720 - pc 0xffffffff88492803 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c8293d0;

   8: fn 0xffffffff883fc0d0 - pc 0xffffffff883fc3a7 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c829460;

   9: fn 0xffffffff883fb970 - pc 0xffffffff883fbbb4 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c829690;

  10: fn 0xffffffff883f86d0 - pc 0xffffffff883f8b70 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c8298e0;

  11: fn 0xffffffff8845e8c0 - pc 0xffffffff8845e8f9 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c829ef0;

  12: fn 0xffffffff8849f8e0 - pc 0xffffffff884cbcf5 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c829f30;

  13: fn 0xffffffff883fa0e0 - pc 0xffffffff883fa16e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c8403d0;

  14: fn 0xffffffff883c1170 - pc 0xffffffff883c1cef / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c840570;

  15: fn 0xffffffff880de7f0 - pc 0xffffffff880de7fe / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c840f90;

  16: fn 0xffffffff88858620 - pc 0xffffffff88858636 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c840fb0;

  17: fn 0xffffffff888333d0 - pc 0xffffffff88833439 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c840fd0;


     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883bdd7; rbp                0x0; rsp 0xffff80007c841000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:113
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 73ctk84we8vz1kw7hv2sr00er:?
(  4) ??:?
(  5) ??:?
(  6) /root/asterinas/kernel/src/process/posix_thread/robust_list.rs:64
(  7) /root/asterinas/kernel/src/process/posix_thread/robust_list.rs:103
(  8) /root/asterinas/kernel/src/process/posix_thread/exit.rs:62
(  9) /root/asterinas/kernel/src/process/posix_thread/exit.rs:36
( 10) /root/asterinas/kernel/src/process/exit.rs:23
( 11) /root/asterinas/kernel/src/syscall/exit_group.rs:13
( 12) /root/asterinas/kernel/src/syscall/mod.rs:157
( 13) /root/asterinas/kernel/src/syscall/mod.rs:328
( 14) /root/asterinas/kernel/src/thread/task.rs:69
( 15) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 16) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 17) /root/asterinas/ostd/src/task/mod.rs:175
make: *** [Makefile:172: run] Error 1
```

--- a/kernel/src/process/posix_thread/robust_list.rs
+++ b/kernel/src/process/posix_thread/robust_list.rs
@@ -43,13 +43,15 @@ impl RobustListHead {
         if self.list_op_pending == 0 {
             None
         } else {
-            Some(self.futex_addr(self.list_op_pending))
+            self.futex_addr(self.list_op_pending)
         }
     }
 
     /// Get the futex address
-    fn futex_addr(&self, entry_ptr: Vaddr) -> Vaddr {
-        (entry_ptr as isize + self.futex_offset) as _
+    fn futex_addr(&self, entry_ptr: Vaddr) -> Option<Vaddr> {
+        self.futex_offset
+            .checked_add(entry_ptr as isize)
+            .map(|result| result as Vaddr)
     }
 }
 
@@ -99,7 +101,7 @@ impl Iterator for FutexIter<'_> {
                 return None;
             }
             let futex_addr = if self.entry_ptr != self.robust_list.list_op_pending {
-                Some(self.robust_list.futex_addr(self.entry_ptr))
+                self.robust_list.futex_addr(self.entry_ptr)
             } else {
                 None
             };

